File: Assets/zfoolua/json/JsonHelloRequest.lua

local JsonHelloRequest = {}

function JsonHelloRequest:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function JsonHelloRequest:protocolId()
    return 1600
end

function JsonHelloRequest:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function JsonHelloRequest:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = JsonHelloRequest:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return JsonHelloRequest

File: Assets/zfoocs/Json/JsonHelloRequest.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class JsonHelloRequest
    {
        public string message;

        public static JsonHelloRequest ValueOf(string message)
        {
            var packet = new JsonHelloRequest();
            packet.message = message;
            return packet;
        }
    }


    public class JsonHelloRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1600;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            JsonHelloRequest message = (JsonHelloRequest) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            JsonHelloRequest packet = new JsonHelloRequest();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}
